rosette/src/BaseSupp.cc

```diff
@@ -153,7 +153,7 @@ int slashify_char(char c, char buf[], int slash_blank) {
             if (isprint(c))
                 buf[1] = c;
             else
-                sprintf(&buf[1], "x%02x", (unsigned char)c);
+                sprintf(&buf[1], "x%02x", (uint8_t)c);
```

Comment:
The original variable "c" is defined as char. The original typecast is unsigned char. Either of these could be more than 8 bits on some systems, but the format clearly is trying to constrain the output to an 8 bit value ("%02x" -- two hex digits). I understand that "%x" is expecting an unsigned int.  If we leave it as some type of char or int, and this turns out to be a larger than 8 bit value, I believe sprintf will emit more than 2 hex digits and I'm not sure if that will be interpreted correctly by the receiver of the result of this function. I used uint8_t because this won't exist on systems without an 8 bit char, and we can deal with it then. FWIW, there was no warning produced.

Reply:
You are correct and I am wrong. It does appear that `printf` functions will output exactly two nibbles when the argument is cast to `(uint8_t)` and more than 2 nibbles when it's an `unsigned int`. Looks to me like the `printf` functions in C++11 are updated to accept `unsigned char` as an argument type, so I guess `uint8_t` works just fine. Good catch.

Reply:
**Done**